src/Npgsql/NpgsqlConnector.cs

```diff
@@ -599,7 +599,7 @@ async Task RawOpen(NpgsqlTimeout timeout, bool async, CancellationToken cancella
         void Connect(NpgsqlTimeout timeout)
         {
             EndPoint[] endpoints;
-            if (!string.IsNullOrEmpty(Host) && Host[0] == '/')
+            if (!string.IsNullOrEmpty(Host) && Directory.Exists(Host))
```

Comment:
> This may be a bit too wide, no? I mean, if the applications happen to run in a directory where there's some file that happens to have the same name as my PostgreSQL server, this code will think that's a Unix-domain socket to connect through...

It actually needs to be a (non-rooted) _directory_ that happens to have the same name as your PostgreSQL server and in that case Npgsql will try to use a file named ".s.PGSQL.5432" inside this directory to connect through. I think that is pretty unlikely to happen but anyways I think that it is a good idea to have a look how psql/libpq behave.
I'll do that.

Reply:
Right, it's a check for a directory. However, as long as the directory exists in the current directory - regardless of whether it has any files in it - it will be understood as a UnixEndPoint and will therefore cause a connection failure, no?

Reply:
You're right.

libpq is actually pretty dumb about this too and uses the same algorithm we used until now.

How about the following?
```csharp
if (!string.IsNullOrEmpty(Host) && (Host[0] == '/' || Host[1] == ':' || Host[0] == '\\'))
```

This would detect any rooted path on both platforms and obviously a non rooted path would probably be a big footgun anyways.

We could also be even more pessimistic and check whether the file exists before we create a `UnixEndPoint`.

Reply:
That looks OK to me. It excludes relative paths to the socket, but there's no way to include those without again conflicting with hostnames. And I don't think anyone cares about relative paths here.